src/coreclr/pal/src/include/pal/context.h

```diff
@@ -384,9 +384,9 @@ bool Xstate_IsApxSupported();
 #define XFEATURE_MASK_AVX512 (XFEATURE_MASK_OPMASK | XFEATURE_MASK_ZMM_Hi256 | XFEATURE_MASK_Hi16_ZMM)
 #endif // XFEATURE_MASK_AVX512
 
-#ifndef XFEATURE_MASK_APX
-#define XFEATURE_MASK_APX (1 << XSTATE_APX)
-#endif // XFEATURE_MASK_APX
+// TODO-xarch-apx: the definition of XSTATE mask value for APX is now missing on the OS level, 
+//                 we are currently using bare value to hack it through the build process, and test the implementation through CI.
+//                 those changes will be removed when we have the OS support for APX.
```

Comment:
Why not do this via the following so that things can still function as expected, even if built on a downlevel OS without the support?
```cpp
#ifndef XSTATE_APX
#define XSTATE_APX 19
#endif
```

Reply:
I tried this way locally on my own dev machine with Windows OS, the build will fail with error report saying the macros are not defined, and it seems that, at least on Windows, `XSTATE_APX` and `XSTATE_MASK_APX` shall be defined in `winnt.h` (I am not 100% sure about the reason, is it because we are using Windows API in the CPUID checks?), I can fix the error by adding the definition there, but it might not be the solution in CI environment.